components/homme/cmake/SetCompilerFlags.cmake

```diff
@@ -67,9 +67,9 @@ IF (${HOMME_USE_CXX})
   SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g")
 
   INCLUDE(CheckCXXCompilerFlag)
-  CHECK_CXX_COMPILER_FLAG("-std=c++14" CXX14_SUPPORTED)
+  CHECK_CXX_COMPILER_FLAG("-std=c++17" CXX14_SUPPORTED)
```

Comment:
here is a typo?

Reply:
@ambrad  or @bartgol -- could i permanently switch to c++17 here?

Reply:
Sure, assuming all the tests pass on all the needed machines. My guess is they will.

Reply:
Does the C++17 requirement come from Crusher or HIP? As far as I can tell, the current Kokkos version is still OK with C++14 for other platforms.

Reply:
I think Kokkos is "close" to switch to C++17 required. IIRC, the next release (3.7, due in Sept) will require it. If it's a deal breaker for HIP, we might think to anticipate kokkos, and switch to C++17 already. I don't think there's a compelling reason to stick with C++14 for the current production clusters, no?

Reply:
i think i needed c++17 for hipcc.